src/framework/theme/components/checkbox/_checkbox.component.theme.scss

```diff
@@ -4,91 +4,106 @@
  * Licensed under the MIT License. See License.txt in the project root for license information.
  */
 
-@mixin checkbox-border($color) {
-  .customised-control-input:checked + .customised-control-indicator {
-    border-color: $color;
-  }
+@mixin nb-checkbox-theme() {
+  nb-checkbox {
+    .custom-checkbox {
+      width: nb-theme(checkbox-width);
+      height: nb-theme(checkbox-height);
+      border-width: nb-theme(checkbox-border-width);
+      border-style: nb-theme(checkbox-border-style);
+      border-radius: nb-theme(checkbox-border-radius);
+    }
+
+    .native-input:focus + .custom-checkbox {
+      box-shadow: 0 0 0 nb-theme(checkbox-outline-width) nb-theme(checkbox-outline-color);
+    }
 
-  &:hover .customised-control-input:enabled + .customised-control-indicator,
-  .customised-control-input:focus:enabled + .customised-control-indicator {
-    border-color: $color;
-    opacity: 0.6;
+    .text {
+      color: nb-theme(checkbox-text-color);
+      font-family: nb-theme(checkbox-text-font-family);
+      font-size: nb-theme(checkbox-text-font-size);
+      font-weight: nb-theme(checkbox-text-font-weight);
+      line-height: nb-theme(checkbox-text-line-height);
+    }
   }
-}
 
-@mixin check-mark-style($size, $color, $border-size) {
-  &::before {
-    height: calc(#{$size} * 0.6);
-    width: calc(#{$size} * 0.4);
-    border-color: $color;
-    border-width: 0 $border-size $border-size 0;
+  @include nb-checkbox-status();
+
+  @each $status in nb-get-statuses() {
+    @include nb-checkbox-status($status);
   }
 }
 
-@mixin nb-checkbox-theme() {
-  nb-checkbox {
-    .customised-control-indicator {
-      background-color: nb-theme(checkbox-bg);
-      width: nb-theme(checkbox-size);
-      height: nb-theme(checkbox-size);
-      border: nb-theme(checkbox-border-size) solid nb-theme(checkbox-border-color);
-
-      @include check-mark-style(
-        nb-theme(checkbox-size),
-        nb-theme(checkbox-checkmark),
-        nb-theme(checkbox-border-size)
-      );
-    }
+@mixin nb-checkbox-status($status: '') {
+  $status: if($status == '', '', -#{$status});
```

Comment:
probably we should not have the `-` part in here `-#{$status}`
and leave it there 
```
background-color: nb-theme(checkbox-#{$status}-background-color);
```
for redability